### ISSUE TITLE
Vibration Action

### ISSUE THREAD

**Issue by GL513** [keymapperorg/KeyMapper#982]
**Developer TODO (don't remove)**
- [ ] create new branch. put issue number at start of name
- [ ] update documentation
- [ ] add button to demo vibration (show button to cancel in case the user inputs a stupid value).
- [ ] sliders for length and intensity.
- [ ] squash and merge
- [ ] delete branch

The action would vibrate the device.
You would he able to customize strength of vibration, and duration (in milliseconds).
It might also be nice to have a pattern picker, or a custon pattern creator using either an easy UI, or a json config file, for example

(vibrate(0(0-0));(1(100-100));(0(0-0)))

Just for a simple 100ms 100% strength buzz
Of course im sure the actual syntax isnt like this, but thats my closest guess.

**Comment by GL513** [comment]
It would also be nice to have a toggle for "demo vibration" next to the one for forcing all keymaps to vibrate, and i think applying the same rules for the action to the force vibrate toggle would be good too.
When the "demo vibration" toggle is on, it will wait two seconds after a vibration value has been edited, and then play the same values back.

**Comment by jambl3r** [comment]
vibrate is now an option instead

**Comment by jambl3r** [comment]
related to #701 
Vibration action/option is to be replaced by a feedback button between actions

**Comment by sds100** [comment]
@jambl3r The vibrate option for key maps currently happens at the start of the action being performed and I think we should keep this behavior. 

Our new design will replace that option. But because in our design we only show the Feedback button _after_ the action, this doesn't really suggest that it happens at the start of the action.

Also, lets say we let users customise the vibration duration between actions etc, do we wait until it finishes before doing the next one? I.e if you make it vibrate for 5 seconds, should it wait for it to finish. A vibration effectively becomes a delay? Or we start the vibration and action in parallel at the same time, and wait for whatever one is longer to finish first? This could be a solution to the comment above.

Maybe we change nothing, or add a chip before the first action (ugly, see screenshot)? I don't really want to put it inside the action. 

<img width="300" alt="Image" src="https://github.com/user-attachments/assets/dcc19bfd-969f-46c2-af3c-3cf15f7bfb47" />

**Comment by sds100** [comment]
We've discussed and we will not implement custom feedback between actions like in the design. The vibrate/on-screen message stays in key map options. We will add a simple vibrate action that lets you specify a custom duration.
